src/ComposerIntegration/BundledPhpExtensionsRepository.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Php\Pie\ComposerIntegration;
+
+use Composer\Package\CompletePackage;
+use Composer\Package\Link;
+use Composer\Package\Package;
+use Composer\Package\Version\VersionParser;
+use Composer\Repository\ArrayRepository;
+use Php\Pie\ExtensionType;
+use Php\Pie\Platform\OperatingSystemFamily;
+use Php\Pie\Platform\TargetPlatform;
+
+use function array_key_exists;
+use function array_map;
+use function sprintf;
+
+class BundledPhpExtensionsRepository extends ArrayRepository
+{
+    /**
+     * @var list<array{
+     *     name: non-empty-string,
+     *     version?: non-empty-string,
+     *     os-families?: non-empty-list<OperatingSystemFamily>,
+     *     type?: ExtensionType,
+     *     priority?: int,
+     * }>
+     */
+    private static array $bundledPhpExtensions = [
+        ['name' => 'bcmath'],
+        ['name' => 'bz2'],
+        ['name' => 'calendar'],
+        [
+            'name' => 'com_dotnet',
+            'os-families' => [OperatingSystemFamily::Windows],
+        ],
+        ['name' => 'ctype'],
+        ['name' => 'curl'],
+        ['name' => 'dba'],
+        ['name' => 'dl_test', 'version' => '>= 8.2.0'],
+        ['name' => 'dom'],
+        ['name' => 'enchant', 'version' => '>= 5.2.0'],
+        ['name' => 'exif'],
+        ['name' => 'ffi', 'version' => '>= 7.4.0'],
+        ['name' => 'gd'],
+        ['name' => 'gettext'],
+        ['name' => 'gmp'],
+        ['name' => 'iconv'],
+        ['name' => 'intl', 'version' => '>= 5.3.0'],
+        ['name' => 'ldap'],
+        ['name' => 'lexbor', 'version' => '>= 8.5.0'],
+        ['name' => 'mbstring'],
+        [
+            'name' => 'mysqli',
+            'priority' => 90, // must load after mysqlnd
+        ],
+        ['name' => 'mysqlnd', 'version' => '>= 5.3.0'],
+        // ['name' => 'odbc'], // build failure - cp: cannot stat '/usr/local/lib/odbclib.a': No such file or directory configure: error: ODBC header file '/usr/local/incl/sqlext.h' not found!
+        [
+            'name' => 'opcache',
+            'type' => ExtensionType::ZendExtension,
+            'version' => '>= 5.5.0',
+        ],
```

Comment:
This extension is always loaded.

Reply:
Sorry, this meant to read "is always loaded with PHP 8.5". This is *not yet* the case, but will likely be with 8.5.0 Alpha 3 as part of https://wiki.php.net/rfc/make_opcache_required.